runtime/zeitgeist/src/xcm_config/fees.rs

```diff
@@ -68,13 +63,22 @@ impl<
 {
     fn get_fee_per_second(location: &MultiLocation) -> Option<u128> {
         let metadata = AssetRegistry::metadata_by_location(location)?;
-        let default_per_second = default_per_second(metadata.decimals);
+        let default_per_second = native_per_second();
+        let native_decimals: u32 = BalanceFractionalDecimals::get().into();
+        let foreign_decimals = metadata.decimals;
+
+        let fee_unadjusted = if let Some(fee_factor) = metadata.additional.xcm.fee_factor {
+            bmul(default_per_second, fee_factor).ok()?
+        } else {
+            default_per_second
+        };
 
-        if let Some(fee_factor) = metadata.additional.xcm.fee_factor {
-            let base = 10u128.checked_pow(metadata.decimals)?;
-            bmul(default_per_second, fee_factor, base)
+        if native_decimals > foreign_decimals {
+            let power = native_decimals - foreign_decimals;
+            Some(fee_unadjusted.checked_div(10u128.checked_pow(power)?)?)
         } else {
-            Some(default_per_second)
+            let power = foreign_decimals - native_decimals;
+            Some(fee_unadjusted.checked_mul(10u128.checked_pow(power)?)?)
```

Comment:
I try to understand. We have cut the decimal places of KSM for example by 10^2 previously. Then we use `default_per_second` multiplied by `fee_factor` (on a native base) to translate the value of ZTG into KSM terms (on a base of ten). And now we need to get back the fee of KSM on a base of twelve for properly charging the KSM XCM fee per second.

Can you give me an example for calculating a `fee_factor` (stored in the native base) for KSM? Step-by-step would be nice. I just try to understand if we not already adjusted the fee enough by using the conversion via the `fee_factor`.

Reply:
The function is applied on the holding register, where the adjustment does not happen. For example:
1 KSM enters and is put into the holding register, holding register has value `10^12`. `fee_per_second` is caluclated by
1. fetching the default fee per second of execution time, which uses the native base representation `10^10`
2. multiplied by `fee_factor`, which uses native base representation  `10^10` (for any token)
3. adjusted to match KSM representation in the holding register, in that case `fee_per_second * 10^2`